devdoc/umock_c_lib_requirements_with_ids.md

```diff
@@ -1202,3 +1202,87 @@ X**SRS_UMOCK_C_LIB_01_195: [** If any error occurs during the destroy_call relat
 
 **SRS_UMOCK_C_LIB_01_196: [** The type used for the return of create_call and first argument of destroy_call shall be allowed to be any type registered with umock. **]**
 XX**SRS_UMOCK_C_LIB_01_204: [** Tracking of paired calls shall not be done if the actual call to the `create_call` is using the `SetFailReturn` call modifier. **]**
+
+## real function support
+
+Sometimes it is desirable to track the mockable function calls and also call the real implementation of those functions.
+While that is possible by simply registering a hook function for each of the mockable functions, it is also rather involved and cumbersome.
+
+In order to simplify the process, the macros `MOCKABLE_INTERFACE`, `IMPLEMENT_MOCKABLE_FUNCTION` and `REGISTER_GLOBAL_INTERFACE_HOOKS` can be used.
+
+Example:
+
+Given a test dependency with the below interface:
+
+```c
+MOCKABLE_INTERFACE(test_interface,
+    FUNCTION(, int, test_dependency_1_arg, int, a),
```

Comment:
>FUNCTION [](start = 4, length = 8)

Is this intended to be MOCKABLE_FUNCTION?

Reply:
here and everywhere

---
In reply to: [293553963](https://github.com/Azure/umock-c/pull/95#discussion_r293553963) [](ancestors = 293553963)

Reply:
Will try to make it MOCKABLE_FUNCTION. There is the other comment below that will remain open for it.

---
In reply to: [293555069](https://github.com/Azure/umock-c/pull/95#discussion_r293555069) [](ancestors = 293555069,293553963)